### PR TITLE
[fix]: [CCM-9902]: Handle Cost Category delete request

### PR DESCRIPTION
**Pull request by iamanmolm**
## Describe your changes
Idea is to find perspective id and name for the given business mapping id list in group by and rules of perspectives. Since later pagination is going to be enabled, so we will have 10 cost categories in a page where we have to make a query only for 10 cost categories. 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40360)

### CODE REVIEW

**Comment by iamanmolm** [comment]
trigger JavaUnitTests9

**Comment by iamanmolm** [comment]
trigger TIAll

File: ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java [hunk 1]
```diff
@@ -136,15 +144,50 @@ public RestResponse<String> update(
...
+      Failsafe.with(transactionRetryPolicy).get(() -> transactionTemplate.execute(status -> {
+        outboxService.save(new CostCategoryDeleteEvent(accountId, costCategory.toDTO()));
+        return true;
+      }));
+    }
+    return new RestResponse<>(perspectiveListMessage);
+  }
+
+  @POST
+  @Path("getPerspectives")
```
[the remark this answers is not in the document]
Comment (iamanmolm) [reply ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java:165]: Done

File: ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java [hunk 2]
```diff
@@ -136,15 +144,50 @@ public RestResponse<String> update(
...
+      responses =
+      {
+        @io.swagger.v3.oas.annotations.responses.
+        ApiResponse(description = "Return list of perspectives corresponding to Business Mapping Id",
+            content = { @Content(mediaType = MediaType.APPLICATION_JSON) })
+      })
+  public RestResponse<List<PerspectiveListMessage>>
+  getPerspectives(@QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) String accountId,
+      @RequestBody(
+          required = true, description = "List of Business Mapping Ids") @Valid List<String> businessMappingIds) {
```
[the remark this answers is not in the document]
Comment (iamanmolm) [reply ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java:185]: Basically raj and I discussed. He will sending list of business mapping id together to get linked perspective in one call itself.
Anyway if he want to call only for one business mapping id, he can send only one element in the list, this exposed API is extensible.

File: ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java [hunk 3]
```diff
@@ -136,15 +144,50 @@ public RestResponse<String> update(
...
+    rbacHelper.checkCostCategoryViewPermission(accountId, null, null);
+    List<PerspectiveListMessage> perspectiveListMessages = null;
+    if (businessMappingIds != null && businessMappingIds.size() > 0) {
```
[the remark this answers is not in the document]
Comment (iamanmolm) [reply ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java:188]: Done

File: 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/CEViewServiceImpl.java [hunk 4]
```diff
@@ -427,6 +428,41 @@ public List<CEView> getViewByState(String accountId, ViewState viewState) {
...
+  }
+
+  @Override
+  public List<PerspectiveListMessage> getViewsByBusinessMapping(String accountId, List<String> businessMappingUuids) {
+    List<PerspectiveListMessage> perspectiveListMessageList = new ArrayList<>();
+    for (String businessMappingUuid : businessMappingUuids) {
+      List<CEView> ceViewList = ceViewDao.findByAccountIdAndBusinessMapping(accountId, businessMappingUuid);
+      PerspectiveListMessage.PerspectiveListMessageBuilder perspectiveListMessageBuilder =
+          PerspectiveListMessage.builder().costCategoryId(businessMappingUuid);
+      if (ceViewList != null && ceViewList.size() > 0) {
```
[the remark this answers is not in the document]
Comment (iamanmolm) [reply 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/CEViewServiceImpl.java:457]: Done

File: 450-ce-views/src/main/java/io/harness/ccm/views/dto/PerspectiveListMessage.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+package io.harness.ccm.views.dto;
+
+import java.util.Map;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class PerspectiveListMessage {
```
[the remark this answers is not in the document]
Comment (iamanmolm) [reply 450-ce-views/src/main/java/io/harness/ccm/views/dto/PerspectiveListMessage.java:16]: Sure

File: 450-ce-views/src/main/java/io/harness/ccm/views/dto/PerspectiveListMessage.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+package io.harness.ccm.views.dto;
+
+import java.util.Map;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
```
[the remark this answers is not in the document]
Comment (iamanmolm) [reply 450-ce-views/src/main/java/io/harness/ccm/views/dto/PerspectiveListMessage.java:15]: Sure, done

File: 450-ce-views/src/main/java/io/harness/ccm/views/dto/PerspectiveListMessage.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (iamanmolm) [reply 450-ce-views/src/main/java/io/harness/ccm/views/dto/PerspectiveListMessage.java:2]: Done

File: 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/CEViewServiceImpl.java [hunk 8]
```diff
@@ -427,6 +428,41 @@ public List<CEView> getViewByState(String accountId, ViewState viewState) {
     return ceViewDao.findByAccountIdAndState(accountId, viewState);
   }
 
+  @Override
+  public PerspectiveListMessage getViewsByBusinessMapping(String accountId, String businessMappingUuid) {
+    List<CEView> ceViewList = ceViewDao.findByAccountIdAndBusinessMapping(accountId, businessMappingUuid);
+    if (ceViewList != null && ceViewList.size() > 0) {
```
[the remark this answers is not in the document]
Comment (iamanmolm) [reply 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/CEViewServiceImpl.java:434]: Done

**Review by sdsingh07 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 6 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40360#-:-NPvqtU0EQtrX1M6qZ6s:bnfp4nl)* status: :shipit: complete! ✓

**Comment by iamanmolm** [comment]
trigger ti5

**Pull request merged**